stunnel/stunnel.ml

```diff
@@ -171,13 +177,16 @@ let config_file verify_cert extended_diagnosis host port =
          ; "ciphers = " ^ Xcp_const.good_ciphersuites
          ; "curve = secp384r1"
          ]
-       ; ( if verify_cert then
+       ; ( match config with
+         | None ->
+             []
+         | Some {sni; cert_bundle_path} ->
              [
                ""
              ; "# use SNI to request a specific cert. CAfile contains"
              ; "# public certs of all hosts in the pool and must contain"
              ; "# the cert of the server we connect to"
-             ; "sni = pool"
+             ; (match sni with None -> "" | Some s -> sprintf "sni = %s" s)
              ; "verifyPeer=yes"
              ; sprintf "CAfile=%s" certificates_bundle_path
              ; ( match Sys.readdir crl_path with
```

Comment:
I've just noticed we lost the checkHost parameter for the appliances:
https://github.com/xapi-project/xen-api-libs-transitional/commit/57dced7a4971722d03a222a419c7e976a1cac4ab#diff-417227ac0f1a2e90d352cebb966c2cd9cdc36e8569b831605c1333ce6398b96aL139

It might be worth using variant for SNI (pool and default) like I did in gencert, and encapsulate the generation of these part of the configuration in a function.

Reply:
So the correct behavior is to use `checkHost` for appliances and `verifyPeer` for SNI?